test/functional/collection_tests.js

```diff
@@ -1581,6 +1581,44 @@ describe('Collection', function() {
     }
   });
 
+  it('should correctly perform estimatedDocumentCount on non-matching query', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ w: 1 }, { poolSize: 1, auto_reconnect: false });
+
+    client.connect(function(err, client) {
+      const db = client.db(configuration.db);
+      let collection;
```

Comment:
You can do `const collection = db.collection('nonexistent_coll_1');` here, and remove those parts from the Promise chain in both tests.